<launch>
  <arg name="dataset" default="$(env PWD)" />
  <arg name="rgbdpair_file" default="$(arg dataset)/assoc.txt" />
  <arg name="groundtruth_file" default="$(arg dataset)/groundtruth.txt" />

  <arg name="launch_dir" value="$(find dvo_benchmark)/launch" />
  <arg name="output_dir" value="$(find dvo_benchmark)/output" />
  <arg name="config_file" default="$(arg launch_dir)/benchmark.yaml" />
  <arg name="slam_config_file" default="$(arg launch_dir)/benchmark_backend.yaml" />
  <arg name="keep_alive" default="false"/>
  
  <!-- if relative path the file will end up in ~/.ros/ -->
  <arg name="trajectory_file" default="$(arg output_dir)/trajectory.txt" />
  <arg name="video_folder" default="$(arg output_dir)/video" />
  <arg name="camera_file" default="$(arg output_dir)/camera.txt" />
  <arg name="hdf5_file" default="$(arg output_dir)/benchmark.h5" />
  <arg name="hdf5_group" default="experiment" />
  <arg name="debug" default="false" />

  <!-- temporary arguments: assignment to parameters -->
  <arg name="sampling_proportion" default="1.0" />
  <arg name="utility_map" default="2" /> <!-- Id, Ramp, Step -->
  <arg name="sampler" default="1" /> <!-- ProbExpected, ProbExact, Deterministic -->
  <arg name="coarsest_level" default="3" /> <!-- 3 or 4 -->
  <arg name="finest_level" default="1" /> <!-- 0 or 1 -->
  <arg name="mu" default="0" /> <!-- Originally 0.05 -->
  <arg name="precision" default="0.0001" /> <!-- Originally 0.0001 -->
  
  <arg name="launch_prefix" value="xterm -e gdb --args" if="$(arg debug)" />
  <arg name="launch_prefix" value="" unless="$(arg debug)" />
  
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(arg launch_dir)/benchmark_logging.config" />
  <env name="ROSCONSOLE_FORMAT"      value="${logger}: { ${message} }" />

  <node launch-prefix="$(arg launch_prefix)" pkg="dvo_benchmark" type="benchmark" name="benchmark" output="screen">
    <param type="string" name="rgbdpair_file" value="$(arg rgbdpair_file)" />
    <param type="string" name="groundtruth_file" value="$(arg groundtruth_file)" />
    
    <param type="bool" name="estimate_trajectory" value="true" />
    <param type="string" name="trajectory_file" value="$(arg trajectory_file)" />

    <param type="string" name="hdf5_file" value="$(arg hdf5_file)" />
    <param type="string" name="hdf5_group" value="$(arg hdf5_group)" />
    
    <param type="bool" name="show_estimate" value="true" />
    <param type="bool" name="show_groundtruth" value="false" />
    <param type="bool" name="keep_alive" value="$(arg keep_alive)" />
    
    <param type="bool" name="render_video" value="false" />
    <param type="string" name="video_folder" value="$(arg video_folder)" />
    
    <param type="string" name="camera_file" value="$(arg camera_file)" />
    
    <rosparam command="load" file="$(arg config_file)" />
    <rosparam command="load" file="$(arg slam_config_file)" />

    <!-- temporary arguments: assignment to parameters -->
    <param type="double" name="sampling_proportion" value="$(arg sampling_proportion)" />
    <param type="int" name="utility_map" value="$(arg utility_map)" />
    <param type="int" name="sampler" value="$(arg sampler)" />
    <param type="int" name="coarsest_level" value="$(arg coarsest_level)" />
    <param type="int" name="finest_level" value="$(arg finest_level)" />
    <param type="double" name="mu" value="$(arg mu)" />
    <param type="double" name="precision" value="$(arg precision)" />
  </node>
</launch>
